drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -414,16 +414,13 @@ static int i2c_mspm0g3xxx_target_register(const struct device *dev,
 		return -ENOTSUP;
 	}
 
-	k_sem_take(&data->i2c_busy_sem, K_FOREVER);
```

Comment:
I've looked into other I2C drivers and they often don't lock un/register callbacks. I'm not sure how I feel about this.
I feel that it's becoming racy.
A solution would be to irq_lock but I'm also not sure how much I like that solution.